2024-11-26 08:05:26,000 DEBUG    root            Loaded Command Group: ['gcloud', 'iam']
2024-11-26 08:05:26,004 DEBUG    root            Loaded Command Group: ['gcloud', 'iam', 'roles']
2024-11-26 08:05:26,066 DEBUG    root            Loaded Command Group: ['gcloud', 'iam', 'roles', 'create']
2024-11-26 08:05:26,070 DEBUG    root            Running [gcloud.iam.roles.create] with arguments: [--permissions: "storage.objects.get,storage.objects.create,pubsub.topics.publish,cloudfunctions.functions.invoke", --project: "inspired-photon-441609-k7", --title: "Custom role for needed services", ROLE_ID: "main_custom_role"]
2024-11-26 08:05:26,146 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:05:26,215 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 403 None
2024-11-26 08:05:58,510 INFO     ___FILE_ONLY___ Enabling service [iam.googleapis.com] on project [644941739395]...

2024-11-26 08:05:58,555 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): serviceusage.googleapis.com:443
2024-11-26 08:05:58,828 DEBUG    urllib3.connectionpool https://serviceusage.googleapis.com:443 "POST /v1/projects/644941739395/services/iam.googleapis.com:enable?alt=json HTTP/1.1" 200 None
2024-11-26 08:05:58,849 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): serviceusage.googleapis.com:443
2024-11-26 08:05:58,955 DEBUG    urllib3.connectionpool https://serviceusage.googleapis.com:443 "GET /v1/operations/acat.p2-644941739395-b1c06da6-14a3-45aa-8d7a-a9abb7927fdf?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:01,684 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): serviceusage.googleapis.com:443
2024-11-26 08:06:01,785 DEBUG    urllib3.connectionpool https://serviceusage.googleapis.com:443 "GET /v1/operations/acat.p2-644941739395-b1c06da6-14a3-45aa-8d7a-a9abb7927fdf?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:01,794 INFO     ___FILE_ONLY___ Operation "operations/acat.p2-644941739395-b1c06da6-14a3-45aa-8d7a-a9abb7927fdf" finished successfully.

2024-11-26 08:06:01,795 DEBUG    root            Request returned no response, retrying
2024-11-26 08:06:01,795 DEBUG    root            Retrying request to url https://iam.googleapis.com/v1/permissions:queryTestablePermissions?alt=json after exception Retry
2024-11-26 08:06:03,687 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:04,188 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:04,286 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:04,663 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:04,747 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:04,803 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 403 None
2024-11-26 08:06:04,809 DEBUG    root            Request returned no response, retrying
2024-11-26 08:06:04,809 DEBUG    root            Retrying request to url https://iam.googleapis.com/v1/permissions:queryTestablePermissions?alt=json after exception Retry
2024-11-26 08:06:06,485 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:06,977 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:07,074 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:07,540 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:07,600 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:07,997 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:08,087 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:08,418 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:08,512 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:08,831 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:08,915 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:09,328 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:09,401 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:09,831 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:09,907 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:10,361 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:10,433 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:10,911 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/permissions:queryTestablePermissions?alt=json HTTP/1.1" 200 None
2024-11-26 08:06:10,927 WARNING  root            API is not enabled for permissions: [pubsub.topics.publish, cloudfunctions.functions.invoke]. Please enable the corresponding APIs to use those permissions.

2024-11-26 08:06:10,930 DEBUG    urllib3.connectionpool Starting new HTTPS connection (1): iam.googleapis.com:443
2024-11-26 08:06:11,026 DEBUG    urllib3.connectionpool https://iam.googleapis.com:443 "POST /v1/projects/inspired-photon-441609-k7/roles?alt=json HTTP/1.1" 403 None
2024-11-26 08:06:11,036 DEBUG    root            (gcloud.iam.roles.create) [644941739395-compute@developer.gserviceaccount.com] does not have permission to access projects instance [inspired-photon-441609-k7] (or it may not exist): You don't have permission to create a role in projects/inspired-photon-441609-k7. This command is authenticated as 644941739395-compute@developer.gserviceaccount.com which is the active account specified by the [core/account] property.
Traceback (most recent call last):
  File "/snap/google-cloud-cli/275/lib/googlecloudsdk/calliope/cli.py", line 998, in Execute
    resources = calliope_command.Run(cli=self, args=args)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/snap/google-cloud-cli/275/lib/googlecloudsdk/calliope/backend.py", line 842, in Run
    resources = command_instance.Run(args)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/snap/google-cloud-cli/275/lib/surface/iam/roles/create.py", line 103, in Run
    result = client.organizations_roles.Create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/snap/google-cloud-cli/275/lib/googlecloudsdk/generated_clients/apis/iam/v1/iam_v1_client.py", line 1126, in Create
    return self._RunMethod(
           ^^^^^^^^^^^^^^^^
  File "/snap/google-cloud-cli/275/lib/third_party/apitools/base/py/base_api.py", line 747, in _RunMethod
    return self.ProcessHttpResponse(method_config, http_response, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/snap/google-cloud-cli/275/lib/third_party/apitools/base/py/base_api.py", line 753, in ProcessHttpResponse
    self.__ProcessHttpResponse(method_config, http_response, request))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/snap/google-cloud-cli/275/lib/third_party/apitools/base/py/base_api.py", line 612, in __ProcessHttpResponse
    raise exceptions.HttpError.FromResponse(
apitools.base.py.exceptions.HttpForbiddenError: HttpError accessing <https://iam.googleapis.com/v1/projects/inspired-photon-441609-k7/roles?alt=json>: response: <{'vary': 'Origin, X-Origin, Referer', 'content-type': 'application/json; charset=UTF-8', 'content-encoding': 'gzip', 'date': 'Tue, 26 Nov 2024 08:06:11 GMT', 'server': 'ESF', 'cache-control': 'private', 'x-xss-protection': '0', 'x-frame-options': 'SAMEORIGIN', 'x-content-type-options': 'nosniff', 'transfer-encoding': 'chunked', 'status': 403}>, content <{
  "error": {
    "code": 403,
    "message": "You don't have permission to create a role in projects/inspired-photon-441609-k7.",
    "status": "PERMISSION_DENIED"
  }
}
>
2024-11-26 08:06:11,060 ERROR    root            (gcloud.iam.roles.create) [644941739395-compute@developer.gserviceaccount.com] does not have permission to access projects instance [inspired-photon-441609-k7] (or it may not exist): You don't have permission to create a role in projects/inspired-photon-441609-k7. This command is authenticated as 644941739395-compute@developer.gserviceaccount.com which is the active account specified by the [core/account] property.
